generators/dart/colour.js

```diff
@@ -11,18 +11,18 @@
 import * as goog from '../../closure/goog/goog.js';
 goog.declareModuleId('Blockly.Dart.colour');
 
-import {dartGenerator, Order} from './dart_generator.js';
+import {Order} from './dart_generator.js';
 
 
-dartGenerator.addReservedWords('Math');
+// RESERVED WORDS: 'Math'
 
-dartGenerator.forBlock['colour_picker'] = function(block, generator) {
+export function colour_picker(block, generator) {
```

Comment:
Because we're just doing `SIMPLE_OPTIMIZATIONS` these won't be renamed by closure correct?

Reply:
Correct.

If we ever get `ADVANCED_OPTIMIZATIONS` working we'd need to rethink this.  One possibility would be to use the functions' built-in `.name` properties when installing them—though to avoid that _also_ being corrupted we might have to resort to writing something like:

```JS
export block_type(/*...*/) {/*...*/};
block_type.name = 'block_type';
```

Option 2B would avoid this potential future problem.

Reply:
If we ever got advanced compilation working, we'd definitely have to have some way of giving closure a list of things not to rename. This is not the first place we'd have an exported function that shouldn't be renamed. But I do not foresee that being in our roadmap anytime soon if ever.